posthog/queries/session_recordings/session_recording_list.py

```diff
@@ -406,3 +435,143 @@ def run(self, *args, **kwargs) -> SessionRecordingQueryResult:
+
+        core_events_query = self._core_events_query.format(
+            properties_select_clause=properties_select_clause,
+            non_aggregate_select_condition_clause=event_filters.non_aggregate_select_condition_clause,
+            event_filter_where_conditions=event_filters.where_conditions,
+            events_timestamp_clause=events_timestamp_clause,
+        )
+
+        grouped_events_query = self._core_events_query_grouped.format(
+            aggregate_event_select_clause=event_filters.aggregate_event_select_clause,
+            ungrouped_core_events_query=core_events_query,
+            event_filter_aggregate_having_clause=event_filters.aggregate_having_clause,
+        )
+
+        return grouped_events_query, {**params, **events_timestamp_params}
+
+    @cached_property
+    def format_event_filters(self) -> EventFiltersSQL:
+        non_aggregate_select_condition_clause = ""
+        aggregate_event_select_clause = ""
+        aggregate_select_clause = ""
+        aggregate_having_clause = ""
+        where_conditions = "AND event IN %(event_names)s"
+        # Always include $pageview events so the start_url and end_url can be extracted
+        event_names_to_filter: List[Union[int, str]] = []
+
+        params: Dict = {}
+
+        for index, entity in enumerate(self._filter.entities):
+            if entity.type == TREND_FILTER_TYPE_ACTIONS:
+                action = entity.get_action()
+                event_names_to_filter.extend([ae for ae in action.get_step_events() if ae not in event_names_to_filter])
+            else:
+                if entity.id not in event_names_to_filter:
+                    event_names_to_filter.append(entity.id)
+
+            condition_sql, filter_params = self.format_event_filter(
+                entity, prepend=f"event_matcher_{index}", team_id=self._team_id
+            )
+            aggregate_event_select_clause += f"""
+            , groupUniqArrayIf(100)((timestamp, uuid, session_id, window_id), event_match_{index} = 1) AS matching_events_{index}
+            , sum(event_match_{index}) AS matches_{index}
+            """
+
+            aggregate_select_clause += f"""
+            , matches_{index}
+            , matching_events_{index}
+            """
+
+            non_aggregate_select_condition_clause += f"""
+            , if({condition_sql}, 1, 0) as event_match_{index}
+            """
+            aggregate_having_clause += f"\nAND matches_{index} > 0"
+            params = {**params, **filter_params}
+
+        params = {**params, "event_names": list(event_names_to_filter)}
+
+        return EventFiltersSQL(
+            non_aggregate_select_condition_clause,
+            aggregate_event_select_clause,
+            aggregate_select_clause,
+            aggregate_having_clause,
+            where_conditions,
+            params,
+        )
+
+    def run(self, *args, **kwargs) -> SessionRecordingQueryResult:
+        self._filter.hogql_context.using_person_on_events = True
```

Comment:
This connection to hogql restricts the improvement to person_on_events enabled orgs for now.

Reply:
don't get this, aren't there like only 4 poe orgs right now?

Reply:
yeah. hogql allows for person properties to be filtered in the recordings event query which usually was not the case. Now, either we use person on events and only have this optimization working for the tiny subset until everything works, or I go through a refactor that splits up and does the right pushdown of person property filtering to the subquery when we detect a hogql based person property 
<img width="545" alt="Screenshot 2023-03-14 at 9 52 03 AM" src="https://user-images.githubusercontent.com/13127476/225059364-a6751566-08ba-4c4f-8a2f-9efa7fa3de9b.png">

Reply:
Ugh I see, okay, good enough